src/test/java/bio/terra/service/resourcemanagement/azure/AzureResourceConfigurationTest.java

```diff
@@ -102,4 +254,149 @@ private BillingProfileModel createProfileModel() {
+                storageAccountName,
+                fileSystemName);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to parse template", e);
+        }
+    }
+
+    private void deleteManagedApplication(AzureResourceManager client,
+                                          ManagedApplicationDeployment applicationDeployment) {
+        logger.info("Deleting the managed application deployment");
+        client.deployments().deleteById(applicationDeployment.applicationDeploymentId);
+
+        logger.info("Deleting the managed application");
+        client.genericResources().deleteById(applicationDeployment.applicationDeploymentId);
+    }
+
+    /**
+     * Stores information relevant to an application's deployment
+     */
+    private static class ManagedApplicationDeployment {
+        // The Azure resource Id
+        private final String applicationDeploymentId;
+        // The name given to the deployment
+        private final String applicationDeploymentName;
+        // The resource group where the application is deployed
+        private final String applicationResourceGroup;
+        // The endpoint to use to connect to the storage account created when deploying the application
+        private final String storageEndpoint;
+        // The name of the storage account
+        private final String storageAccountName;
+        // The name of the filesystem created when deploying the application
+        private final String fileSystemName;
+
+        ManagedApplicationDeployment(String applicationDeploymentId,
+                                     String applicationDeploymentName,
+                                     String applicationResourceGroup,
+                                     String storageEndpoint,
+                                     String storageAccountName,
+                                     String fileSystemName) {
+            this.applicationDeploymentId = applicationDeploymentId;
+            this.applicationDeploymentName = applicationDeploymentName;
+            this.applicationResourceGroup = applicationResourceGroup;
+            this.storageEndpoint = storageEndpoint;
+            this.storageAccountName = storageAccountName;
+            this.fileSystemName = fileSystemName;
+        }
+    }
+
+    private String generateSas(DataLakeFileClient client) {
+        OffsetDateTime startTime = OffsetDateTime.now();
+        OffsetDateTime expiryTime = startTime.plusHours(1);
+        FileSystemSasPermission permission = new FileSystemSasPermission().setReadPermission(true)
+            .setWritePermission(false);
+
+        // Note: can't use user delegation since that's not supported cross tenant
+        // Note: the version is so that subdirectory sharing works...docs say it's a no-op, but it sure isn't...
+        return client.generateSas(new DataLakeServiceSasSignatureValues(expiryTime, permission)
+            .setStartTime(startTime).setVersion("2020-04-08"));
```

Comment:
According to the Azure docs, if we don't set a start time the token will be valid immediately so we could take this out: https://docs.microsoft.com/en-us/azure/storage/common/storage-sas-overview (under the "Be careful with SAS start time" section)

Reply:
Also where does that version date come from? Is it random?

Reply:
Ah great question...that `2020-04-08` doesn't refer to a date but a version of the Sas API (which is not the latest but is compatible with certain permissions we'll need).

Good point...we could also take out the start time.

Reply:
Got it! Thanks for clarifying